smc_trading_agent/v1/routers/auth.py

```
@@ -0,0 +1,56 @@
+
+    user = User(email=payload.email, password_hash=hash_password(payload.password))
+    db.add(user)
+    db.add(AuditLog(action="user.register", detail=f"Registered {payload.email}"))
+    db.commit()
+
+    token = create_access_token(payload.email)
+    return TokenResponse(access_token=token)
+
+
+@router.post("/login", response_model=TokenResponse)
+def login(payload: UserLogin, db: Session = Depends(get_db)):
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)